### PR TITLE
Homework4: Character device creation 

### PR DESCRIPTION
**Pull request by kalydmysfs**
Created character device driver, /proc and /sys interfaces,  and testing bash script.

### CODE REVIEW

File: Device/char_Mydriver [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	is_open = 0;
+	used_size = 0;
+
+	if(alloc_size < DEFALT_BUFFER_SIZE) {
+		printk("Buffer size must be at least %d bytes!\n", DEFALT_BUFFER_SIZE);
+		goto buffer_too_small;
+	}
+
+	cdev_init(&dev_struct, &fops);
+	cdev_add(&dev_struct, dev_num, NUMBER_OF_DEVICES);
```
Comment (oleg-h2o) [inline Device/char_Mydriver]: cdev_add() can return an error. Please handle it.

File: Device/char_Mydriver [hunk 2]
```diff
@@ -0,0 +1,130 @@
+#include <linux/uaccess.h>	//copy_to_user, copy_from_user
```
Comment (oleg-h2o) [inline Device/char_Mydriver]: This file should have .c extension, isn't it?

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Plz make those small fixes.

File: Device/char_Mydriver (hunk 2 above)
Comment (KalynenkoDima) [reply Device/char_Mydriver]: I think that we have to use #include <linux/uaccess.h>becouse this file.h include function prototype and after that we write implementation of our function whitch name is copy_to_user.

File: Device/char_Mydriver (hunk 1 above)
Comment (kalydmysfs) [reply Device/char_Mydriver]: Done. please check @oleg-h2o

File: Device/char_Mydriver (hunk 2 above)
Comment (kalydmysfs) [reply Device/char_Mydriver]: I think that we have to include  <linux/uaccess.h> , becouse this file.h include 
function prototype сopy_to_user but concrete implementation i did in <linux/uaccess.c>

Comment (galkinletter) [reply Device/char_Mydriver]: Need char_mydriver.c

This file should have .c extension,

File: Device/char_Mydriver [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	printk("Driver removed!\n");
+}
+
+module_init(init);
+module_exit(exit);
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Dmytro Kalynenko  <dmytrokalynenko@gmail.com>");
+MODULE_DESCRIPTION("Simple character device driver");
+MODULE_VERSION("0.1");
```
Comment (galkinletter) [inline Device/char_Mydriver]:
```suggestion
MODULE_VERSION("0.1");


```

Please fix end of line

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix

File: Device/char_Mydriver (hunk 2 above)
Comment (KalynenkoDima) [reply Device/char_Mydriver]: Done.

File: Device/char_Mydriver (hunk 3 above)
Comment (kalydmysfs) [reply Device/char_Mydriver]: Done. please check @galkinletter

File: Device/char_Mydriver (hunk 2 above)
Comment (kalydmysfs) [reply Device/char_Mydriver]: Done. please check @galkinletter

**Comment by galkinletter** [comment]
OK

**Pull request merged**